src/InlineReductions.h

```diff
@@ -36,6 +36,7 @@ namespace Halide {
  */
 //@{
 Expr sum(Expr, const std::string &s = "sum");
+Expr saturating_sum(const Expr &init_val, Expr e, const std::string &s = "saturating_sum");
```

Comment:
Thinking about it you are right. `init_val` is not required, I will revert it. When I first wrote this function I had the expression `saturating_sum(u8(f(x)) * i8(f(x))` in mind. When using groups of 4 8 bit integers and a 32bit accumulator you can never saturate the result unless `init_val` is something close to `INT_MAX` but I didn't consider the general case where this won't map directly to the vnni instructinos. I guess this ties in with my question https://github.com/halide/Halide/discussions/5825, since inline reductions are basically a shortcut for some common expression patterns.